libsovtoken/tests/payment_chaos_tests.rs

```diff
@@ -0,0 +1,357 @@
+extern crate libc;
+extern crate sovtoken;
+extern crate indy;                      // lib-sdk project
+
+#[macro_use] extern crate log;
+#[macro_use] extern crate serde_derive;
+#[macro_use] extern crate serde_json;
+
+pub mod utils;
+
+use indy::ErrorCode;
+use sovtoken::logic::config::output_mint_config::MintRequest;
+use sovtoken::logic::request::Request;
+use sovtoken::utils::base58::IntoBase58;
+use sovtoken::utils::random::rand_bytes;
+use utils::setup::{Setup, SetupConfig};
+use utils::wallet::Wallet;
+
+// ------------------------------------------------------------------------------------------------
+const VERKEY_LEN: usize = 32;
+
+// ------------------------------------------------------------------------------------------------
+pub fn gen_random_base58_verkey() -> String {
+    let vk_bytes = rand_bytes(VERKEY_LEN);
+    vk_bytes.into_base58()
+}
+
+// ------------------------------------------------------------------------------------------------
+pub fn do_minting(pool_handle : i32, wallet : &Wallet, dids : &Vec<&str>, payment_address : &str, amount : i32) {
+    let mint_output_json = json!([
+        {
+            "recipient": payment_address,
+            "amount": amount,
+        }
+    ]).to_string();
+
+    let (mint_req, _) = indy::payments::Payment::build_mint_req(
+        wallet.handle,
+        Some(&dids[0]),
+        &mint_output_json,
+        None,
+    ).unwrap();
+
+    trace!("{:?}", &mint_req);
+
+    let mint_req = Request::<MintRequest>::multi_sign_request(
+        wallet.handle,
+        &mint_req,
+        dids.clone()
+    ).unwrap();
+
+    trace!("{:?}", &mint_req);
+
+    let _ = indy::ledger::Ledger::submit_request(pool_handle, &mint_req).unwrap();
+}
+
+// ------------------------------------------------------------------------------------------------
+#[test]
+pub fn pay_without_outputs_fails() {
+
+    // ---- setup
+    sovtoken::api::sovtoken_init();
+    let wallet = Wallet::new();
+    let setup = Setup::new(&wallet, SetupConfig {
+        num_addresses: 1,
+        num_trustees: 4,
+        num_users: 0,
+        mint_tokens: None,
+        fees: None,
+    });
+    let payment_addresses = &setup.addresses;
+    let pool_handle = setup.pool_handle;
+    let dids = setup.trustees.dids();
+    let tokens : i32 = 50;
+
+    // ---- create some tokens
+    do_minting(pool_handle, &wallet, &dids, &payment_addresses[0], tokens);
```

Comment:
You can do minting as part of the Setup/SetupConfig, so you don't need to have this do_minting function.